### PR TITLE
Chore: Changes to support edge connector

### LINKED ISSUE #432: Need to add type definitions to npm package so we can create external rule/connector
If we try to create an external rule/connector, we don't have the type definitions, so typescript notifies some errors when they are not.

For example, if we try to create a new `Launcher` for a connector, I have to do something like:
`export class NewLauncher exports Launcher`

`Launcher` has a protected property `port`, but if I try to use that property inside `NewLauncher` I get a typescript error `error TS2339: Property 'port' does not exist on type 'NewLauncher'.`

### PR DESCRIPTION
**Pull request by sarvaje**
This PR has to be merged to make https://github.com/sonarwhal/edge15-connector works

Also, it would be great to have fixed #432 so we can change [this code](https://github.com/sonarwhal/edge15-connector/blob/master/src/lib/connectors/edge15-connector/edge15-connector-launcher.ts#L42)

### CODE REVIEW

File: src/lib/connectors/shared/remote-debugging-connector.ts [hunk 1]
```diff
@@ -73,6 +74,7 @@ export class Connector implements IConnector {
         const defaultOptions = {
             loadCompleteRetryInterval: 250,
             maxLoadWaitTime: 30000,
+            newTabUrl: 'http://sonarwhal-empty-site.azurewebsites.net/',
```
Comment (alrra) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: @sarvaje Maybe add a bit of explanation why this URL is used, who owns it, etc. Thanks!

Comment (molant) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: It will actually be empty.sonarwhal.com once I'm back full time 😊

Comment (sarvaje) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: >It will actually be empty.sonarwhal.com once I'm back full time 😊

I'll change the url.

>@sarvaje Maybe add a bit of explanation why this URL is used, who owns it, etc. Thanks!

ok!

File: src/lib/connectors/shared/async-html.ts [hunk 2]
```diff
@@ -48,7 +54,15 @@ export class AsyncHTMLDocument implements IAsyncHTMLDocument {
     }
 
     public async pageHTML(): Promise<string> {
-        const { outerHTML } = await this._DOM.getOuterHTML({ nodeId: this._dom.nodeId });
+        let { outerHTML } = await this._DOM.getOuterHTML({ nodeId: this._dom.nodeId });
+
+        // Some browsers like Edge doesn't have the property outerHTML in the root element
```
Comment (molant) [inline src/lib/connectors/shared/async-html.ts]: > don't

File: src/lib/connectors/shared/remote-debugging-connector.ts [hunk 3]
```diff
@@ -73,6 +74,10 @@ export class Connector implements IConnector {
         const defaultOptions = {
             loadCompleteRetryInterval: 250,
             maxLoadWaitTime: 30000,
+            // tabUrl is a empty html site used to avoid edge diagnostics adapter to receive unexpeted onLoadEventFired
+            // and onRequestWillBeSent events from the default url opened when you create a new tab in Edge.
+            tabUrl: 'http://empty.sonarwhal.com/',
```
Comment (molant) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: Let's use `https://empty.sonarwhal.com`
I'll configure the SSL certificate once the DNS records are working.

File: src/lib/rules/html-checker/html-checker.ts [hunk 4]
```diff
@@ -78,7 +78,7 @@ const rule: IRuleBuilder = {
                when testing the rule and `import` doesn't work here. */
             const htmlChecker = require('html-validator');
 
-            scanOptions.data = response.body.content;
+            scanOptions.data = response.body.rawContent.toString();
```
Comment (molant) [inline src/lib/rules/html-checker/html-checker.ts]: I don't think this change is related to edge 15. Maybe put it as a different commit?

File: src/lib/connectors/shared/remote-debugging-connector.ts [hunk 5]
```diff
@@ -338,6 +341,52 @@ export class Connector implements IConnector {
...
+            await this._server.emitAsync('manifestfetch::error', event);
+        }
+    }
+
+    private async getManifest(element: IAsyncHTMLElement) {
+        this._manifestIsSpecified = true;
+
+        try {
+            // CDP will not download the manifest on its own, so we have to "force" it
+            // This will trigger the `onRequestWillBeSent`, `OnResponseReceived`, and
```
Comment (molant) [inline src/lib/connectors/shared/remote-debugging-connector.ts]: > `onResponseReceived`

File: src/lib/rules/html-checker/html-checker.ts (hunk 4 above)
Comment (sarvaje) [inline src/lib/rules/html-checker/html-checker.ts]: It is related to edge15. IIRC, rawContent in edge15 is in base64 and a string in chrome

Comment (molant) [inline src/lib/rules/html-checker/html-checker.ts]: What? Then you should put a comment about this.

**Comment by sarvaje** [comment]
@molant everything should be done!

**Comment by molant** [comment]
@alrra do you want any other changes or can this be merged?
